packages/cevitxe-storage-abstract/src/index.ts

```diff
@@ -5,18 +5,18 @@ export abstract class StorageAdapter {
 
   constructor(options: { discoveryKey: string; databaseName: string }) {
     const { databaseName, discoveryKey } = options
-    this.storageKey = `cevitxe::${databaseName}::${discoveryKey.substr(0, 12)}`
+    this.storageKey = `cevitxe_${databaseName}_${discoveryKey.substr(0, 12)}`
   }
 
   abstract async open(): Promise<void>
   abstract async close(): Promise<void>
   abstract async hasData(): Promise<boolean>
 
-  abstract get changes(): AsyncIterableIterator<IteratorResult<ChangeSet, ChangeSet>>
+  abstract changes(): AsyncIterableIterator<ChangeSet>
   abstract async getChanges(documentId: string): Promise<ChangeSet[]>
   abstract async appendChanges(changeSet: ChangeSet): Promise<void>
 
-  abstract get snapshots(): AsyncIterableIterator<IteratorResult<SnapshotRecord, SnapshotRecord>>
+  abstract snapshots(): AsyncIterableIterator<SnapshotRecord>
```

Comment:
Instead of iterating over a cursor and then having to take its `value`, we'll iterate over the thing itself. 

Also, can't have a property that's a generator, so this needs to be a method.

Reply:
see implementations: 
https://github.com/DevResults/cevitxe/pull/59/files#diff-109994a59530d76838c782b2e30975abL41-L51
https://github.com/DevResults/cevitxe/pull/59/files#diff-194c0b388b49de5901864229782217ebR45-R54